python/desc/imsim/skyModel.py

```diff
@@ -14,32 +14,46 @@
 import galsim
 from lsst.sims.GalSimInterface.galSimNoiseAndBackground import NoiseAndBackgroundBase
 
+from lsst.sims.photUtils import BandpassDict
+
+from lsst.sims.photUtils import Sed
+
 from .imSim import get_config
 
 __all__ = ['skyCountsPerSec', 'ESOSkyModel', 'get_skyModel_params']
 
+#galBandpassDict = BandpassDict.loadTotalBandpassesFromFiles(['u','g', 'r', 'i', 'z', 'y'])
 
-# Code snippet from D. Kirkby.  Note the use of astropy units.
-def skyCountsPerSec(surface_brightness=21, filter_band='r',
-                    effective_area=32.4*u.m**2, pixel_size=0.2*u.arcsec):
-    pars = get_skyModel_params()
-    # Lookup the zero point corresponding to 24 mag/arcsec**2
-    s0 = pars[filter_band] * u.electron / u.s / u.m ** 2
-
-    # Calculate the rate in detected electrons / second
-    dB = (surface_brightness - pars['B0']) * u.mag(1 / u.arcsec ** 2)
-    return s0 * dB.to(1 / u.arcsec ** 2) * pixel_size ** 2 * effective_area
 
+def skyCountsPerSec(skyModel, photPars, filter_name='u', magNorm=None):
+    if not hasattr(skyCountsPerSec, '_bp_dict'):
+    	skyCountsPerSec._bp_dict = BandpassDict.loadTotalBandpassesFromFiles(['u','g', 'r', 'i', 'z', 'y'])
+    bandpass = skyCountsPerSec._bp_dict[filter_name]
+    wave, spec = skyModel.returnWaveSpec()
+    skymodel_Sed = Sed(wavelen=wave, flambda=spec[0, :])
+    if magNorm:
+        skymodel_fluxNorm = skymodel_Sed.calcFluxNorm(magNorm, bandpass)
+        skymodel_Sed.multiplyFluxNorm(skymodel_fluxNorm)
+    sky_counts = skymodel_Sed.calcADU(bandpass=bandpass, photParams=photPars) * (
+        100**2) / photPars.exptime / photPars.effarea / photPars.nexp / photPars.gain * u.electron / u.s / u.m ** 2
+    return sky_counts
 
 # Here we are defining our own class derived from NoiseAndBackgroundBase for
 # use instead of ExampleCCDNoise
+
+
 class ESOSkyModel(NoiseAndBackgroundBase):
     """
     This class wraps the GalSim class CCDNoise.  This derived class returns
     a sky model based on the ESO model as implemented in
     """
 
-    def __init__(self, obs_metadata, seed=None, addNoise=True, addBackground=True):
+    def __init__(
```

Comment:
It would be good to avoid format changes like this that are non-idiomatic for this repo, especially if the original code already follows the nominal coding conventions (e.g., pep8).